src/main/java/core/basesyntax/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage;
+
+    public BalanceOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void count(FruitTransaction transaction) {
+        if (transaction.getQuantity() < 0) {
+            throw new RuntimeException("Quantity can't be less than 0");

```

Comment:
It's good that you're checking for negative quantity, but consider moving this validation logic to a separate method or class that is responsible for validating `FruitTransaction` objects. This will make the `count` method more focused on its primary responsibility and adhere better to the single-responsibility principle.